snuba/utils/metrics/backends/dualwrite.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import annotations
+
+import random
+
+from sentry_sdk import metrics
+
+from snuba import settings, state
+from snuba.utils.metrics.backends.abstract import MetricsBackend
+from snuba.utils.metrics.backends.datadog import DatadogMetricsBackend
+from snuba.utils.metrics.backends.sentry import SentryMetricsBackend
+from snuba.utils.metrics.types import Tags
+
+
+class SentryDatadogMetricsBackend(MetricsBackend):
+    """
+    A metrics backend that records metrics to Sentry and Datadog.
+    """
+
+    def __init__(
+        self, datadog: DatadogMetricsBackend, sentry: SentryMetricsBackend
+    ) -> None:
+        self.datadog = datadog
+        self.sentry = sentry
+
+    def _use_sentry(self) -> bool:
+        if state.get_config("use_sentry_metrics", "0") == "1":
+            return bool(random.random() < settings.DDM_METRICS_SAMPLE_RATE)
+        return False
+
+    def increment(
+        self,
+        name: str,
+        value: int | float = 1,
+        tags: Tags | None = None,
+        unit: str | None = None,
+    ) -> None:
+        self.datadog.increment(name, value, tags, unit)
+        if self._use_sentry():
+            self.sentry.increment(name, value, tags, unit)
+
+    def gauge(
+        self,
+        name: str,
+        value: int | float,
+        tags: Tags | None = None,
+        unit: str | None = None,
+    ) -> None:
+        self.datadog.gauge(name, value, tags, unit)
+        if self._use_sentry():
+            self.sentry.gauge(name, value, tags, unit)
+
+    def timing(
```

Comment:
timings are distributions -- i don't think we need both. the implementation in the SDK is identical

Reply:
They're not identical, because in the SDK they accept different units. The timing ones specifically only accept timing units, and distributions are more flexible.

Reply:
can we have a default impl for timing that uses distribution internally or the other way around? The impl really is identical in the SDK (the only thing that differs is type hints and return value) -- i get that you want to have different defaults for unit in each case

Reply:
Why not both? Both exist in the Sentry SDK and in Datadog. Why not use the SDKs as intended?

Reply:
because in both SDKs they're literally just aliases and now you're forcing every backend to implement those aliases.

the purpose of `sentry_sdk.metrics.timing` is to provide a decorator and context manager to measure a codeblock -- we're not using that

Reply:
> forcing every backend to implement those aliases.

This is true, but we are missing the `distribution` functionality from all our backends, so I would have to add that no matter what (timing is a subset of distribution). Why not leave timing in place, add distributions, and then our backends line up with both of the production SDKs (DD and Sentry).

> literally just aliases

If the aliases are already available, why not leverage them? Why in turn make timing _another_ alias on distribution? The DD and Sentry SDKs already provide that alias for us.

Reply:
> I would have to add that no matter what

you really don't, you could continue to use timing or rename it to distribution if you want to. timing is not a subset of distribution in master branch, it's just distribution with maybe an unfortunate (not-technically-correct) name. i don't see the point in adding more methods that do the same thing but only differ in their default for unit, but I don't feel too strongly about all of this so i'm approving